java/src/main/native/src/TableJni.cpp

```diff
@@ -935,13 +935,77 @@ jlongArray combine_join_results(JNIEnv *env, cudf::table &left_results,
   return combine_join_results(env, std::move(left_cols), std::move(right_cols));
 }
 
+cudf::column_view remove_validity_from_col(cudf::column_view column_view) {
+  std::unique_ptr<cudf::column_view> ret;
+  if (!cudf::is_nested(column_view.type())) {
```

Comment:
```suggestion
  if (!cudf::is_compound(column_view.type())) {
```
The `is_compound()` will be true for all the nested types as well as string (and dictionary). So you don't have special code for string here.